torch/distributed/_tensor/dispatch.py

```diff
@@ -230,6 +236,21 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
             redistribute_with_schema=needs_redistribute,
         )
 
+        aten = torch.ops.aten
+        random_ops = [
+            aten.native_dropout.default,
+            aten.normal_.default,
+            aten.uniform_.default,
+        ]
+        # before running local op computation, check if op is random op
+        # for random ops, set RNG offset
+        if op_call in random_ops:
```

Comment:
I think we probably need to add a private API in `random.py`, sth like `is_rng_supported_mesh` and add the condition to here and the one below. This is so that we don't crash on other device mesh types (i.e. cpu). We can raise warning inside `is_rng_supported_mesh` if it's not a cuda device mesh.

Reply:
good catch! I just realized that we need to check types here as well after seeing `test_init` failed.